src/backend/app/auth/auth_routes.py

```diff
@@ -127,45 +128,89 @@ async def logout():
 async def get_or_create_user(
     db: Session,
     user_data: AuthUser,
-) -> DbUser:
+):
     """Get user from User table if exists, else create."""
-    existing_user = await user_crud.get_user(db, user_data.id)
-
-    if existing_user:
-        # Update an existing user
-        if user_data.img_url:
-            existing_user.profile_img = user_data.img_url
-        db.commit()
-        return existing_user
-
-    user_by_username = await user_crud.get_user_by_username(db, user_data.username)
-    if user_by_username:
-        raise HTTPException(
-            status_code=400,
-            detail=(
-                f"User with this username {user_data.username} already exists. "
-                "Please contact the administrator."
-            ),
+    try:
+        update_sql = text(
+            """
+            INSERT INTO users (
+                    id, username, profile_img, role, mapping_level,
+                    is_email_verified, is_expert, tasks_mapped, tasks_validated,
+                    tasks_invalidated, date_registered, last_validation_date
+                    )
+                VALUES (
+                    :user_id, :username, :profile_img, :role,
+                    :mapping_level, FALSE, FALSE, 0, 0, 0,
+                    :current_date, :current_date
+                    )
+            ON CONFLICT (id)
+                DO UPDATE SET profile_img = :profile_img;
+            """
         )
+        role = UserRole(user_data.role).name
+        db.execute(
+            update_sql,
+            {
+                "user_id": user_data.id,
+                "username": user_data.username,
+                "profile_img": user_data.profile_img,
+                "role": role,
+                "mapping_level": "BEGINNER",
+                "current_date": datetime.now(timezone.utc),
+            },
+        )
+        db.commit()
 
-    # Add user to database
-    db_user = DbUser(
-        id=user_data.id,
-        username=user_data.username,
-        profile_img=user_data.img_url,
-        role=user_data.role,
-    )
-    db.add(db_user)
-    db.commit()
-
-    return db_user
-
-
-@router.get("/me/", response_model=AuthUser)
+        get_sql = text(
+            """
+            SELECT users.*,
+                COALESCE(user_roles.project_id) as project_id,
+                COALESCE(user_roles.role, 'MAPPER') as project_role,
+                COALESCE(organisation_managers.organisation_id) as created_org
+            FROM users
+            LEFT JOIN user_roles ON users.id = user_roles.user_id
+            LEFT JOIN organisation_managers on users.id = organisation_managers.user_id
+            WHERE users.id = :user_id;
+            """
+        )
+        result = db.execute(
+            get_sql,
+            {"user_id": user_data.id},
+        )
+        db_user = result.fetchall()
```

Comment:
.first() will return a single value, so that you don't have to index with [0] below. Then you also can avoid needing the for loop too 👍

Reply:
Yes, you are right. Great! 
Thanks for the feedback!!